### ISSUE TITLE
The sphinx-autogen example is broken

### ISSUE THREAD

**Issue by danking** [sphinx-doc/sphinx#4724]
### Problem

After trying to make sphinx-autogen work in an existing project to no success, I decided to attempt to make the [sphinx-autogen example work](http://www.sphinx-doc.org/en/1.7/man/sphinx-autogen.html). AFAICT, it does not work as written with sphinx-autogen 1.7.0.

#### Procedure to reproduce the problem
```
mkdir -p docs
mkdir -p foobar/foo
mkdir -p foobar/bar/baz
touch foobar/foo/__init__.py
touch foobar/bar/__init__.py
touch foobar/bar/baz/__init__.py
cat > docs/index.rst <<EOF
Modules
=======

.. autosummary::
   :toctree: modules

   foobar.foo
   foobar.bar
   foobar.bar.baz
EOF
sphinx-autogen docs/index.rst
```

NB: the sphinx-autogen docs suggest using the command line `sphinx-autodoc doc/index.rst` I changed the command to `sphinx-autogen` (I don't have anything called `sphinx-autodoc` on my path) and the path to `docs/index.rst`.

#### Error logs / results
```
[autosummary] generating autosummary for: docs/index.rst
WARNING: [autosummary] failed to import u'foobar.bar': no module named foobar.bar
WARNING: [autosummary] failed to import u'foobar.bar.baz': no module named foobar.bar.baz
WARNING: [autosummary] failed to import u'foobar.foo': no module named foobar.foo
```

The docs folder now contains a folder called "modules" (before running the command no such folder was present), but it is empty.
```
dking@wmb16-359 # ls -al docs/modules
total 0
drwxr-xr-x  2 dking  wheel   68 Mar  8 14:13 .
drwxr-xr-x  4 dking  wheel  136 Mar  8 14:13 ..
```

#### Expected results

The example advertises the following:

> then the following stub files will be created in docs:
```
docs
├── index.rst
└── modules
    ├── foobar.bar.rst
    ├── foobar.bar.baz.rst
    └── foobar.foo.rst
```

### Environment info
- OS: Mac OS X Version 10.12.6 (16G1212)
- Python version:
- Sphinx version:
```
dking@wmb16-359 # python --version
Python 3.6.4 :: Anaconda, Inc.
dking@wmb16-359 # sphinx-build --version
sphinx-build 1.7.0
```

**Comment by tk0miya** [comment]
You're right.
It is needed to add loading path of python using `PYTHONPATH` or other ways.

**Comment by tk0miya** [comment]
Fixed by #4762 
Thank you for reporting!
